src/test/java/teammates/test/cases/action/StudentCourseJoinAuthenticatedActionTest.java

```diff
@@ -26,14 +27,16 @@ protected String getActionUri() {
     }
 
     @BeforeClass
-    public void classSetup() {
+    public void classSetup() throws Exception {
         // extra test data used on top of typical data bundle
-        removeAndRestoreDataBundle(dataBundle);
     }
 
     @Override
     @Test
     public void testExecuteAndPostProcess() throws Exception {
+        dataBundle = loadDataBundle("/StudentCourseJoinAuthenticatedTest.json");
+        removeAndRestoreDataBundle(dataBundle);
+
```

Comment:
load of dataBundle is required here as it is executed after access control and in accessControl we have changes dataBundle to `typicaldataBundle`.
May we can remove Line 22 and add data bundle here